### PR TITLE
home-assistant-custom-components.spook: init at 3.0.1

### PR DESCRIPTION
**Pull request by kkoniuszy**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - Tested this with my home-assistant instance
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mweinelt** [comment]
If it ships two components, can we expose them as two dedicated component packages, sharing the same src?

**Comment by kkoniuszy** [comment]
> If the repo contains two components, can we expose them as two dedicated component packages, sharing the same src?

@mweinelt They aren't supposed to be installed separately, in HACS there's just a single Spook package. `spook_inverse` is [a helper](https://spook.boo/inverse) that's meant to be a part of `spook`. The separate manifest.json is just a hack that makes it possible to provide a helper – it wouldn't be possible otherwise, as the `integration_type: "hub"` in the core manifest does not let the component provide helpers.

It's just kind of messy. The HACS version does this weird thing where it symlinks `spook_inverse` to `custom_components` then restarts hass, which I patched out here. Separating it into two packages is not the intended use, but it's technically possible as the child component doesn't really depend on any code from the parent. Should I try do that or keep it reasonably close to upstream?

File: pkgs/servers/home-assistant/custom-components/spook/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    fnv-hash-fast
+    psutil-home-assistant
+    sqlalchemy
+  ];
+
+  meta = {
+    changelog = "https://github.com/frenck/spook/releases/tag/v${version}";
+    description = "Toolbox for Home Assistant";
+    homepage = "https://spook.boo/";
+    license = "MIT";
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/spook/default.nix:35]:
```suggestion
    license = lib.licenses.mit
```

File: pkgs/servers/home-assistant/custom-components/spook/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = domain;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ChHsevryWuim8BEFqXVkCOW9fGMrt5vol+B2SreMUws=";
+  };
+
+  patches = [./remove-sub-integration-symlink-hack.patch];
+
+  propagatedBuildInputs = [
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/spook/default.nix:24]:
```suggestion
  dependencies = [
```

Newer lingo with the same effect.

**Comment by mweinelt** [comment]
@ofborg test home-assistant

**Comment by mweinelt** [comment]
Thanks!

**Pull request merged**